### ISSUE TITLE
add feature to hook global key bindings into (desktop?) OS

### ISSUE THREAD

**Issue by yangguangwuwu** [fyne-io/fyne#2304]
add Handle Key Presses in App

hi!  I want to listen for shortcuts in the background and then do some tasks

I looked at the documentation and didn't find the API

**Comment by Jacalz** [comment]
Please use the templates (the feature request template in this case) when opening issues in the future.

There is already support for this. You should be able to pass a function to the `.SetOnTypedKey()` method on the window canvas (https://pkg.go.dev/fyne.io/fyne/v2#Canvas). That should be your window `.Canvas().SetOnTypedKey()`.

**Comment by yangguangwuwu** [comment]
I'm sorry 
I wrote a new one
.Canvas().SetOnTypedKey()       This function is inconsistent



### Is your feature request related to a problem? Please describe:

yes 
I need to achieve a quick screenshot through the keyboard
Minimize background running
.Canvas()   .Canvas().SetOnTypedKey()  .Canvas().SetOnTypedKey() 
.Canvas() .AddShortcut(shortcut Shortcut, handler func(shortcut Shortcut))
Only when the program is running in the foreground can it take effect
Only when the current work window takes effect
### Is it possible to construct a solution with the existing API?

no
Canvas()   .Canvas().SetOnTypedKey()  .Canvas().SetOnTypedKey() 
.Canvas() .AddShortcut()  
Only when the current work window takes effect
### Describe the solution you'd like to see:

I need an API
API similar to  .Canvas() .AddShortcut()  
background running
Non current work window takes effect

**Comment by Jacalz** [comment]
The function is not inconsistent, it is simply not supposed to take inputs other than when the canvas is focused. Most operating systems, as far as I know, don't exactly send key presses to applications when they are not focused. That usually requires some kind of deeper hook into the key handling in the desktop environment and that is going to be very platform specific. I think that it most likely is outside the scope of Fyne, but that is just my understanding of the matter.

**Comment by Jacalz** [comment]
If I understand your use case correctly, I think the most sensible solution in this case would be to overwrite a shortcut in your operating system to run your application instead.

**Comment by yangguangwuwu** [comment]
I think there is something wrong with my description, which makes you understand it wrong

For example

I'm playing a game

I use the keyboard shortcut of OBS Studio software to record video in the game

 .Canvas() .AddShortcut()   Cannot be performed in inactive window 

My active window is the game interface

**Comment by yangguangwuwu** [comment]
Fyne lacks API to implement function

These hotkeys can be run in inactive windows


![微信截图_20210701101738](https://user-images.githubusercontent.com/24450871/124054992-aded0880-da55-11eb-8766-39a0143fed62.png)

**Comment by yangguangwuwu** [comment]
几乎很多应用都有这种非活动窗口接收快捷键的功能 

OBS discord ....

**Comment by andydotxyz** [comment]
The implementation is not inconsistent, it simply lacks the feature you desire - OS global
Keys must be registered separately and have complications such as needing to be globally unique. I have updated the title accordingly.

**Comment by yangguangwuwu** [comment]
thank you

**Comment by andydotxyz** [comment]
Reopening to track the feature addition.

**Comment by netrip** [comment]
@Jacalz thank your quick response! Is there any plan to deliver this feature?

**Comment by netrip** [comment]
Many thanks!

**Comment by onyx-and-iris** [comment]
I would very much like to see the ability to register global hotkeys as well. I'm would like to develope a 'hotkeys' type application to run in the background with other windows in focus. I'll keep an eye on this thread.

**Comment by matwachich** [comment]
Seems like this feature is coming to GLFW
https://github.com/glfw/glfw/issues/1204

**Comment by dpengchen** [comment]
Is that now possible?

**Comment by Jacalz** [comment]
No. The GLFW issue has not been resolved yet and this issue is also open still.

**Comment by ivancho-ifa** [comment]
I managed to work around it using https://github.com/golang-design/hotkey
